heka/cmd/heka-s3list/main.go

```diff
@@ -0,0 +1,98 @@
+/***** BEGIN LICENSE BLOCK *****
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# The Initial Developer of the Original Code is the Mozilla Foundation.
+# Portions created by the Initial Developer are Copyright (C) 2015
+# the Initial Developer. All Rights Reserved.
+#
+# Contributor(s):
+#   Mark Reid (mark@mozilla.com)
+# ***** END LICENSE BLOCK *****/
+
+/*
+
+A command-line utility for listing files on Amazon S3, filtered by dimension.
+
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/crowdmob/goamz/aws"
+	"github.com/crowdmob/goamz/s3"
+	"github.com/mozilla-services/data-pipeline/heka/plugins/s3splitfile"
+	"os"
+	"strings"
+	"time"
+)
+
+func main() {
+	flagSchema := flag.String("schema", "", "Filename of the schema to use as a filter")
+	flagBucket := flag.String("bucket", "default-bucket", "S3 Bucket name")
+	flagBucketPrefix := flag.String("bucket-prefix", "", "S3 Bucket path prefix")
+	flagAWSKey := flag.String("aws-key", "DUMMY", "AWS Key")
+	flagAWSSecretKey := flag.String("aws-secret-key", "DUMMY", "AWS Secret Key")
+	flagAWSRegion := flag.String("aws-region", "us-west-2", "AWS Region")
+	flagDryRun := flag.Bool("dry-run", false, "Don't actually do anything, just output what would be done")
+	flagVerbose := flag.Bool("verbose", false, "Print detailed info")
+	flag.Parse()
+
+	if flag.NArg() != 0 {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	var err error
+	var schema s3splitfile.Schema
+	schema, err = s3splitfile.LoadSchema(*flagSchema)
+	if err != nil {
+		fmt.Printf("schema: %s\n", err)
+		os.Exit(2)
+	}
+
+	if *flagDryRun {
+		fmt.Printf("Dry Run: Would have listed files in s3://%s/%s according to filter schema %s\n",
+			*flagBucket, *flagBucketPrefix, *flagSchema)
+		os.Exit(0)
+	}
+
+	var b *s3.Bucket
+
+	prefix := strings.Trim(*flagBucketPrefix, "/")
+	if prefix != "" {
+		prefix += "/"
+	}
+
+	// Initialize the S3 bucket
+	auth := aws.Auth{AccessKey: *flagAWSKey, SecretKey: *flagAWSSecretKey}
+	s := s3.New(auth, aws.Regions[*flagAWSRegion])
```

Comment:
The `cat` handling will give you a sensible error message if you get the parameter wrong. I will update this one to match.